plugins/editqml/qml/PaletteControls.qml

```diff
@@ -86,6 +95,86 @@ QtObject{
 
     //////////////////////////////////////////////
 
+    property QtObject views: QtObject {
+
+        function openPaletteListBox(style, model, editorBoxParams, handlers){
+            var paletteList = createPaletteListView(null, style)
+            paletteList.model = model
+
+            var paletteListBox = lk.layers.editor.environment.createEditorBox(
+                paletteList,
+                editorBoxParams.aroundRect,
+                editorBoxParams.panePosition,
+                editorBoxParams.relativePlacement
+            )
+
+            paletteListBox.color = 'transparent'
+            paletteList.forceActiveFocus()
+
+            if (handlers){
+                paletteList.cancelledHandler = function(){
+                    handlers.onCancelled(paletteListBox)
+                }
+                paletteList.selectedHandler = function(index){
+                    handlers.onSelected(index, paletteListBox)
+                }
+            }
+            return paletteList
+        }
+
+        function openPaletteListForNode(container, paletteGroup, parent){
```

Comment:
We need 2 functions:

```js
openPaletteList(...)
openPaletteListBox(...)
```

```openPaletteList``` is missing.
```openPaletteList``` should be called from `openPaletteListBox`

```openPaletteList``` simply creates a paletteList view, without a palleteListBox wrapper around. It can then be used for the node function.

```js
paletteControls.views.openPaletteList(
    model,
    style,
    {
        onSelected: function(view, index, palette){ ... },
        onCanceled: function(view){ ... }
    },
    parent
}
```

```js
paletteControls.views.openPaletteListBox(
    aroundBox,
    Qt.point(paneCoords.x, paneCoords.y - 35),
    lk.layers.editor.environment.placement.bottom,
    style,
    model,
    {
        onSelected: function(box, index, palette){ ... },
        onCanceled: function(box){ ... }
    }
}
```